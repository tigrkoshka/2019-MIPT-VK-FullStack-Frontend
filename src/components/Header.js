import React from 'react'
import { Link } from 'react-router-dom'
import PropTypes from 'prop-types'
import headerStyles from '../styles/headerStyles.module.scss'
import imagesStyles from '../styles/imagesStyles.module.scss'

function Header({ leftLink, leftImg, rightText, rightImg, name, onRightClick }) {
  let backName
  let rightButton
  if (leftLink.substring(0, 9) === '/ChatList') {
    backName = (
      <div className={headerStyles.vertical}>
        <div className={headerStyles.text}>Chats</div>
      </div>
    )
  } else if (leftLink === '/') {
    backName = (
      <div className={headerStyles.vertical}>
        <div className={headerStyles.text}>Home</div>
      </div>
    )
  }
  if (rightImg !== '') {
    rightButton = (
      <div className={headerStyles.rightButton} onClick={onRightClick}>
        <div className={headerStyles.vertical}>
          <div
            className={`${imagesStyles[rightImg]}`}
            style={{
              height: '35px',
              width: '35px',
            }}
          />
        </div>
      </div>
    )
  } else if (rightText !== '') {
    rightButton = (
      <div className={headerStyles.rightButton} onClick={onRightClick}>
        <div className={headerStyles.vertical}>
          <div className={headerStyles.text}>{rightText}</div>
        </div>
      </div>
    )
  }
  return (
    <div className={headerStyles.rectangle}>
      <div className={headerStyles.leftHorizontal}>
        <Link to={leftLink} className={headerStyles.leftButton}>
          <div className={headerStyles.vertical}>
            <div
              className={`${imagesStyles[leftImg]}`}
              style={{
                height: '25px',
                width: '25px',
              }}
            />
          </div>
          {backName}
        </Link>
      </div>
      <div className={headerStyles.horizontal}>
        <div className={headerStyles.vertical}>
          <div className={headerStyles.name}>{name}</div>
        </div>
      </div>
      <div className={headerStyles.rightHorizontal}>{rightButton}</div>
    </div>
  )
}

Header.propTypes = {
  leftLink: PropTypes.string.isRequired,
  leftImg: PropTypes.string.isRequired,
  rightText: PropTypes.string.isRequired,
  rightImg: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  onRightClick: PropTypes.func.isRequired,
}

export default Header
