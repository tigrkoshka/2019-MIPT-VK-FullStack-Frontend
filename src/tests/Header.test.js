/* eslint-disable react/jsx-props-no-spreading */
import React from 'react'
import { MemoryRouter } from 'react-router-dom'
import renderer from 'react-test-renderer'
import Header from '../components/Header'
import toChats from '../images/back.png'

describe('Header renders and acts correctly', () => {
  let component

  beforeEach(() => {
    component = renderer
      .create(
        <MemoryRouter>
          <Header
            leftImg={toChats}
            rightImg=""
            rightText=""
            leftLink="/ChatList/1"
            name="Test header"
            onRightClick={() => {}}
          />
        </MemoryRouter>,
      )
      .toJSON()
  })

  it('Should render normally', () => {
    expect(component).toMatchSnapshot()
  })
})
