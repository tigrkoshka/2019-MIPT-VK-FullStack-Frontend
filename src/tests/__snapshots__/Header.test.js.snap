// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders and acts correctly Should render normally 1`] = `
<div
  className="rectangle"
>
  <div
    className="leftHorizontal"
  >
    <a
      className="leftButton"
      href="/ChatList/1"
      onClick={[Function]}
    >
      <div
        className="vertical"
      >
        <img
          alt=""
          height="25px"
          src="back.png"
          width="height"
        />
      </div>
      <div
        className="vertical"
      >
        <div
          className="text"
        >
          Chats
        </div>
      </div>
    </a>
  </div>
  <div
    className="horizontal"
  >
    <div
      className="vertical"
    >
      <div
        className="name"
      >
        Test header
      </div>
    </div>
  </div>
  <div
    className="rightHorizontal"
  />
</div>
`;
